rdrf/rdrf/admin.py

```diff
@@ -178,6 +183,93 @@ def get_readonly_fields(self, request, obj=None):
         return () if obj is None else ("code",)
 
 
+class ActivationKeyExpirationListFilter(admin.SimpleListFilter):
+    title = _('activation key expired')
+    parameter_name = 'activation_key_expired'
+
+    def lookups(self, request, model_admin):
+        return [
+            ('True', _('True')),
+            ('False', _('False')),
+        ]
+
+    def queryset(self, request, queryset):
+        expired_emails = []
+        unexpired_emails = []
+        for profile in queryset:
+            user_email = profile.user.email
+            if profile.activation_key_expired():
+                expired_emails.append(user_email)
+            else:
+                unexpired_emails.append(user_email)
+
+        if self.value() == 'True':
+            return RegistrationProfile.objects.filter(user__email__in=expired_emails)
```

Comment:
chain to existing `queryset` rather than creating a new filter from the `RegistrationProfile`.